discovery_engine_core/core/src/config.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::sync::Arc;
+
+use figment::{
+    providers::{Format, Json, Serialized},
+    Figment,
+};
+use serde::{Deserialize, Serialize};
+use tokio::{join, sync::RwLock};
+
+use xayn_discovery_engine_ai::CoiSystemConfig;
+use xayn_discovery_engine_providers::Market;
+
+/// Configuration settings to initialize Discovery Engine with a
+/// [`xayn_discovery_engine_ai::Ranker`].
+#[derive(Clone, Debug)]
+pub struct InitConfig {
+    /// Key for accessing the API.
+    pub api_key: String,
+    /// API base url.
+    pub api_base_url: String,
+    /// List of markets to use.
+    pub markets: Vec<Market>,
+    /// List of trusted sources to use.
+    pub trusted_sources: Vec<String>,
+    /// List of excluded sources to use.
+    pub excluded_sources: Vec<String>,
+    /// S-mBert vocabulary path.
+    pub smbert_vocab: String,
+    /// S-mBert model path.
+    pub smbert_model: String,
+    /// KPE vocabulary path.
+    pub kpe_vocab: String,
+    /// KPE model path.
+    pub kpe_model: String,
+    /// KPE CNN path.
+    pub kpe_cnn: String,
+    /// KPE classifier path.
+    pub kpe_classifier: String,
+    /// AI config in JSON format.
+    pub ai_config: Option<String>,
```

Comment:
❔  Maybe we should rename it because it's now not just the ai config anymore? what about `user_config` or `custom_config`?

Reply:
i agree, i also thought about renaming this, but was lacking a good name. wdyt about `de_config` as we can configure many things in the whole DE. or maybe `json_config` to point at the config format?

Reply:
`de_config` sounds good to me

Reply:
ah good point 👍  thanks for checking